service/dap/daptest/client.go

```diff
@@ -97,83 +99,9 @@ func (c *Client) ExpectUnsupportedCommandErrorResponse(t *testing.T) *dap.ErrorR
 	return c.expectErrorResponse(t, 9999, "Unsupported command")
 }
 
-func (c *Client) ExpectDisconnectResponse(t *testing.T) *dap.DisconnectResponse {
-	t.Helper()
-	return c.ExpectMessage(t).(*dap.DisconnectResponse)
-}
-
-func (c *Client) ExpectContinueResponse(t *testing.T) *dap.ContinueResponse {
-	t.Helper()
-	return c.ExpectMessage(t).(*dap.ContinueResponse)
-}
-
-func (c *Client) ExpectNextResponse(t *testing.T) *dap.NextResponse {
-	t.Helper()
-	return c.ExpectMessage(t).(*dap.NextResponse)
-}
-
-func (c *Client) ExpectStepInResponse(t *testing.T) *dap.StepInResponse {
-	t.Helper()
-	return c.ExpectMessage(t).(*dap.StepInResponse)
-}
-
-func (c *Client) ExpectStepOutResponse(t *testing.T) *dap.StepOutResponse {
-	t.Helper()
-	return c.ExpectMessage(t).(*dap.StepOutResponse)
-}
-
-func (c *Client) ExpectTerminatedEvent(t *testing.T) *dap.TerminatedEvent {
-	t.Helper()
-	return c.ExpectMessage(t).(*dap.TerminatedEvent)
-}
-
-func (c *Client) ExpectInitializeResponse(t *testing.T) *dap.InitializeResponse {
-	t.Helper()
-	initResp := c.ExpectMessage(t).(*dap.InitializeResponse)
-	if !initResp.Body.SupportsConfigurationDoneRequest {
```

Comment:
This check got lost in the shuffle. You need a wrapper around the other helper to test for this and other capabilities we expect to be set.

Reply:
Thanks. Nice catch. Added as ExpectInitializeResponseAndCapability (yeah, not a great name)

Curious why this capability check should be done for every initialize response - will the dap server change the set of supported capabilities across testing?

Reply:
It's fixed now. This might change with rr because it might or might not be supported on different architectures. We could just test this explicitly in some tests, but not others. It seemed easier to just hide this in a helper, even if that means rechecking the same (minor) thing every time.